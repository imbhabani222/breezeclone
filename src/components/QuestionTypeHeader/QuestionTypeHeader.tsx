import React, { useEffect, useState, useCallback } from "react";
import {
  Space,
  Typography,
  Menu,
  Dropdown,
  Button,
  message,
  Select,
  Switch,
} from "antd";
import DropdownArrow from "../../assets/dropdown_btn.svg";
import lastSaveClock from "../../assets/clock_save.svg";

import style from "./questionTypeHeader.module.scss";

import ExclamationCircleDark from "../../assets/exclamation_circle_dark.svg";
import ChangeThemeIcon from "../../assets/change-theme.svg";

import threeDots from "../../assets/ThreeDots.svg";

import loadStub from "../../assets/loadStub.svg";
import toggle from "../../assets/toggleSvg.svg";

import UnsolveIcon from "../../assets/unsolve-icon.svg";
import mobile_edit from "../../assets/mobile_edit_circle.svg";

import mcq_edit from "../../assets/mcqedit.svg";

import moment from "moment";
import clsx from "clsx";
import { useSelector } from "react-redux";
import { getTechnology } from "../../selectors/selectors";

const { Text } = Typography;

const QuestionTypeHeader = (props: any) => {
  const { showLastSave } = props;
  const [language, setLanguage] = useState(props.language);
  const [expires, setExpires] = useState<any>(null);
  const getTime = (time: string) => {
    return moment(time).format(" h:mm a");
  };

  const onTechnologyChange = (event: any) => {
    props.handleLanguage(event);
  };

  useEffect(() => {
    setExpires(moment.utc(props.lastSave).valueOf());
  }, [props.lastSave]);
  useEffect(() => {
    setLanguage(props.language);
  }, [props.language]);

  const statusData = [
    {
      icon: props.title === "UNSOLVED" ? UnsolveIcon : mcq_edit,
      text: props.title,
    },
    {
      icon: !props.lastSave ? lastSaveClock : lastSaveClock,
      className:
        props.lastSave === "No Data" ? style.hideSave : "showLastSaveIcon",
      text: expires ? `Last save at ${getTime(expires)}` : "No Data",
      hide: props?.title === "UNSOLVED" ? !showLastSave : showLastSave,
    },
  ];

  const changeTheme: any = () => {
    props.changeTheme();
  };

  const { Option } = Select;

  // let defaultLang = Object.keys(props.defaultLangauges);

  const getTechnologies = useSelector(getTechnology);

  let getAllowedTech = getTechnologies;
  let allowedTechnologies = getAllowedTech?.technologies?.filter((val: any) =>
    val.allowed_in.includes("SCR")
  );
  let mapSlugdata = allowedTechnologies?.map((tech: any) => tech);

  // let stubdata = allowedTechnologies?.map((tech: any) => tech);

  // console.log(stubdata, "stub");
  let currentLanguage = null || "Node";

  function onChange(checked: any) {
    console.log(`switch to ${checked}`);
  }

  const menu = (
    <Menu>
      <div className={style.dropDownContainer}>
        <Menu.Item>
          <div className={style.menuItem}>
            <img src={loadStub} alt={"ICON"} />
            <p className={style.content}> Load Language Stub</p>
          </div>
        </Menu.Item>
        {/* <div className={style.menuItem}> */}
        {/* <Menu.Item>
          <img src={toggle} alt={"ICON"} className={style.toggle} />
          Enable auto-complete
        </Menu.Item> */}
        {/* </div> */}
      </div>
    </Menu>
  );

  // console.log(props, "props");

  return (
    <div className={style.container}>
      <div className={style.problemType}>
        <div className={style.heading}>
          {<img src={ExclamationCircleDark} alt={"Icon"} />}
          <h4>{props.type}</h4>
        </div>
      </div>
      {props.theme ? (
        <div className={style.selectDropdown}>
          <Select
            suffixIcon={<img src={DropdownArrow} alt={"Icon"} />}
            value={language}
            onChange={onTechnologyChange}
          >
            {mapSlugdata &&
              mapSlugdata?.map((lang: any, index: number) => {
                return (
                  <Option value={lang.slug} key={index}>
                    {lang.name}
                  </Option>
                );
              })}
          </Select>
        </div>
      ) : null}
      <div className={style.status}>
        <>
          {statusData
            .filter((item: any) => !item.hide)
            .map((item: any, idx) => (
              <div
                key={idx}
                className={clsx(
                  props.title === "UNSOLVED"
                    ? style.statusItem
                    : style.statusItemWhite,
                  item.className
                )}
              >
                <img src={item.icon} alt={"ICON"} />
                <Text>{item.text}</Text>
              </div>
            ))}
          {props.theme ? (
            <>
              <div className={style.changeTheme}>
                <Button className={style.themeCssolor} onClick={changeTheme}>
                  <img src={ChangeThemeIcon} alt={"ICON"} />
                  Change Theme
                </Button>
              </div>

              <div className={style.changeTheme}>
                {/* <Button className={style.themeCssolor} onClick={changeTheme}>
                  :
                </Button> */}
                <Dropdown overlay={menu} placement="bottomRight" arrow>
                  <Button>
                    <img src={threeDots} alt={"ICON"} />
                  </Button>
                </Dropdown>
              </div>
            </>
          ) : null}
        </>
      </div>
    </div>
  );
};

export default React.memo(QuestionTypeHeader);
