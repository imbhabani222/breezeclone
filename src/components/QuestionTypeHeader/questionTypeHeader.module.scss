@import "../../commonStyles/color.module.scss";
@import "../../commonStyles/button.module.scss";
@import "../../commonStyles/font.module.scss";
@import "../../commonStyles/media-queries.scss";
.container {
  height: 50px;
  background: $coding-linenum-clr;
  @include dFlex(center, space-between);
  padding: 10px 10px 10px 30px;
  @include respond(medium) {
    // height: 120px;
    // @include dFlex(flex-start, space-between);
    // flex-direction: column;
    // padding: 10px 0px 20px 15px;
  }
  // @include respond(medium) {
  //   padding: 10px;
  // }
  @include respond(small) {
    height: 100px;
    flex-wrap: wrap;
    padding: 0px;
  }
  .problemType {
    @include dFlex(center, center);
    margin: 12px 0;
    @include respond(small) {
      margin: 0px;
    }
  }
  .heading {
    @include dFlex(center, center);
    img {
      @include respond(small) {
        display: none;
      }
    }
    .qus_round {
      display: none;
      @include respond(small) {
        display: block;
      }
    }
    h4 {
      padding: 0 10px;
      margin-bottom: 0;
      @include fontStyle(21px, 700, $question-header-heading);
      @include respond(max_mid) {
        @include fontStyle(19px, 700, $question-header-heading);
      }
      @include respond(small) {
        padding: 0 23px;
        @include fontStyle(15px, 700, $question-header-heading);
      }
    }
  }
  .langBtn {
    button {
      border-radius: 5px;
      border: 1px solid $editor-dropdown;
      background: $white-color;
      height: 40px;
      span:first-child {
        padding-right: 90px;
      }
    }
  }

  .status {
    @include dFlex(center, center);

    @include respond(medium) {
      // width: 100%;
      justify-content: space-between;
      // margin-top: 20px;
    }
    @include respond(small) {
      width: 100%;
      justify-content: space-between;
      margin-top: 0;
      padding: 5px 15px;
      //box-shadow: 0px 1px 2px rgba(117, 117, 117, 0.25);
    }
    div:nth-child(2) {
      background-color: $white-color !important;
      span {
        color: $invalid-btn-text !important;
      }
    }
    .statusItemWhite :global {
      background-color: $white-color;
      @include dFlex(center, center);
      margin: auto 8px;
      border-radius: 16px;
      padding: 2px 15px 2px 6px;
      img {
        margin-right: 7px;
      }
      .ant-typography {
        @include fontStyle(14px, 700, $question-header-heading);
        word-wrap: break-word;
        @include respond(small) {
          @include fontStyle(13px, 700, $question-header-heading);
        }
      }
    }

    .statusItem :global {
      background-color: $status-btn-gray;
      @include dFlex(center, center);
      margin: auto 28px;
      border-radius: 16px;
      padding: 2px 15px 2px 6px;
      img {
        margin-right: 7px;
      }
      .ant-typography {
        @include fontStyle(14px, 400, $question-header-heading);
        color: $white-color;
        word-wrap: break-word;
        @include respond(small) {
          @include fontStyle(13px, 400, $white-color);
        }
      }
    }
    .themeColor :global {
      display: none;
      margin-right: 20px;
      img {
        margin-left: 5px;
      }
      .ant-typography {
        @include fontStyle(14px, 500, $change-theme-clr);
        margin-left: 5px;
        @include respond(small) {
          display: none;
        }
      }
    }
  }
}

.hideSave {
  display: none !important;
}

.changeTheme :global {
  [ant-click-animating-without-extra-node="true"]::after {
    display: none;
  }
  .ant-btn:hover,
  .ant-btn:focus {
    background-color: transparent !important;
  }
}

.changeTheme button :global {
  background-color: transparent;
  color: $black-color;
  border: none;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: -4px;
  img {
    margin: 0 10px;
  }
}
.selectDropdown :global {
  // width: 170px;
  // margin-right: 133px;
  margin-right: -120px;
  position: fixed;
  left: 701px;

  background: transparent;
  border-color: transparent;
  .ant-select-focused:not(.ant-select-disabled).ant-select:not(.ant-select-customize-input)
    .ant-select-selector {
    border-color: transparent;
    box-shadow: none;
  }
  .ant-select :global {
    height: 40px;
    padding: 3px;
    width: 180px !important;
    background: $white-color;
    box-sizing: border-box;
    border: 1px solid $editor-dropdown;
    border-radius: 4px;
    margin-left: 98px;
  }
  .ant-select:not(.ant-select-customize-input) .ant-select-selector {
    border: 0;
    border-color: transparent;
    box-shadow: none;
    // border-radius: 5px;
  }
}

.dropDownContainer {
  padding: 10px;
  // .content {
  //   padding-left: 10px;
  // }
  .menuItem {
    display: flex;

    .content {
      margin-left: 19px;
      margin-top: 11px;
    }
  }

  .toggle {
    margin-right: 15px;
  }
}
