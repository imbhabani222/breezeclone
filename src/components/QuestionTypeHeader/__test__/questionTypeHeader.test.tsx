import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import QuestionTypeHeader from "../QuestionTypeHeader";

describe("Button Group Testing", () => {
  //   const handleClick = jest.fn();
  //   const defaultProps: ButtonProps = {
  //     className: "Clear",
  //     display: "Clear",
  //     type: "submit",
  //     onClick: handleClick,
  //     beforeIcon: "beforeIcon",
  //     afterIcon: "afterIcon",
  //     disabled: true,
  //   };

  //   const ButtondataProps: ButtonArrayProps = {
  //     buttons: [defaultProps],
  //     className: "footer_btn_div",
  //   };
  function renderQuestionTypeHeader() {
    return render(<QuestionTypeHeader />);
  }

  afterEach(cleanup);

  test("Render without Crashing", () => {
    renderQuestionTypeHeader();
  });
  test("Match snapshot QuestionTypeHeader component", () => {
    const { asFragment } = renderQuestionTypeHeader();
    expect(asFragment()).toMatchSnapshot();
  });
});
