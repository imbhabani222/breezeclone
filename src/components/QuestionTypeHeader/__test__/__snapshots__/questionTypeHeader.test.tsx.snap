// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Group Testing Match snapshot QuestionTypeHeader component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="problemType"
    >
      <div
        class="heading"
      >
        <img
          alt="Icon"
          src="exclamation_circle_dark.svg"
        />
        <h4>
          MCQ
        </h4>
      </div>
    </div>
    <div
      class="status"
    >
      <div
        class="statusItemWhite"
      >
        <img
          alt="ICON"
          src="mcqEdit.png"
        />
        <span
          class="ant-typography"
        />
      </div>
      <div
        class="statusItemWhite"
      >
        <img
          alt="ICON"
          src="clock_save.svg"
        />
        <span
          class="ant-typography"
        >
          Last save  3:19 pm
        </span>
      </div>
      <div
        class="themeColor"
      >
        <img
          alt="ICON"
          src="change-theme.svg"
        />
        <span
          class="ant-typography"
        >
          Change Theme
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
